training/MAXTEXT_README.md

```diff
@@ -8,36 +8,41 @@ JAX < 0.7.0.
 ```shell
 git clone https://github.com/google/maxtext.git
 cd maxtext
-# Checkout either the commit id or MaxText tag. 
-# Example: `git checkout tpu-recipes-v0.1.2`
+
+# Specify MaxText commit id or tag you want to checkout
+# You can find MaxText tag here: https://github.com/AI-Hypercomputer/maxtext/tags
+MAXTEXT_COMMIT_ID_OR_TAG=#<commit_id_or_tag> e.g. tpu-recipes-v0.1.2
 git checkout ${MAXTEXT_COMMIT_ID_OR_TAG}
 ```
 
 2. Install MaxText dependencies
 ```shell
-bash setup.sh
+# Install uv, a fast Python package installer
+pip install uv
+
+# Install MaxText and its dependencies
+uv pip install maxtext --resolution=lowest
+install_maxtext_github_deps
 ```
 
 Optional: Use a virtual environment to setup and run your workloads. This can help with errors
 like `This environment is externally managed`:
 ```shell
 ## One time step of creating the venv
-VENV_DIR=~/venvp3
-python3 -m venv $VENV_DIR
-## Enter your venv.
-source $VENV_DIR/bin/activate
-## Install dependencies
-bash setup.sh
+VENV_NAME=#<your_virtual_env_name>
+pip install uv
+uv venv --python 3.12 --seed $VENV_NAME
+source $VENV_NAME/bin/activate
+
+# Install MaxText and its dependencies
+uv pip install -e .[tpu] --resolution=lowest
+install_maxtext_github_deps
 ```
 
-> **_NOTE:_** If you use a virtual environment, you must use the same one when running the 
-[XPK Installation](https://github.com/AI-Hypercomputer/xpk?tab=readme-ov-file#installation) 
-steps linked in the [XPK_README](XPK_README.md) as well as your relevant tpu-recipe workloads.
-
 3. Run the following commands to build the docker image
 ```shell
 # Example BASE_IMAGE=us-docker.pkg.dev/cloud-tpu-images/jax-stable-stack/tpu:jax0.5.2-rev1
-BASE_IMAGE=<stable_stack_image_with_desired_jax_version>
+BASE_IMAGE=#<stable_stack_image_with_desired_jax_version>
```

Comment:
Note that long-term we are moving off JAX AI Image in favor of pinning transitive dependencies in `bash docker_build_dependency_image.sh DEVICE=tpu MODE=stable`